docs/source/transaction_family_specifications/validator_registry_transaction_family.rst

```diff
@@ -0,0 +1,257 @@
+information is used to decide what number of blocks a validator has to wait
+for before it can start claiming blocks after it adds new signup information
+to the validator registry. This check is necessary because if the number of
+blocks that must be waited on is greater than the number of validators minus
+one, it is possible for the network to get into a state where nobody can
+publish blocks because the validators are all waiting for more blocks to be
+committed or their signup information to be added to a block.
+
+Validator registry transaction would not be able to be done at the same time as
+any other transaction as an update to the ValidatorMap is necessary. However all
+other transaction that need to access the state set by the validator registry,
+can be done in parallel since it will only be a read and  the statics for each
+validator is stored in the poet_enclave. If this was changed so that the stats
+were stored in the validator registry this would require a write to state every
+time a block is published and would reduce the ability for parallelism.
+
+Addressing
+----------
+
+When a validator’s signup info is registered or updated it should be accessed
+using the following algorithm:
+
+Addresses for the validator registry transaction family are set by adding
+sha256 hash of the validator's id to the validator registry namespace. The
+namespace for the validator registry will be the first 6 characters of the
+sha256  hash of the string “validator_registry”, which is “6a4372” For example,
+the validator signup info of a validator with the id “validator_id” could be
+set like this:
+
+.. code-block:: pycon
+
+  >>>“6a4372” + hashlib.sha256('validator_id'.encode("utf-8")).hexdigest()
+  '6a43722aee5b550a3cbd1595f4de10049ee805bc035b5e232dfacfc31cc6275170b30d'
+
+The map of current the current registered validator_id should be stored at the
+ following address:
+
+
+.. code-block:: pycon
+
+  >>>“6a4372” + hashlib.sha256('validator_map'.encode()).hexdigest()
+  '6a437247a1c12c0fb03aa6e242e6ce988d1cdc7fcc8c2a62ab3ab1202325d7d677e84c'
+
+Transaction Payload
+===================
+Validator registry transaction family payloads are defined by the following
+protocol buffers code:
+
+.. code-block:: protobuf
+
+  message ValidatorRegistryPayload {
+    // The action that the transaction processor will take. Currently this
+    // is only “register”, but could include other actions in the futures
+    // such as “revoke”
+  	string verb = 1;
+
+    // The name of the endpoint. It should be a human readable name.
+  	string name = 2;
+
+
+  	// validator's public key (currently using signer_pubkey as this is
+    // stored in the transaction header)
+      string id = 3;
+
+    // This is the protocol buffer described above.
+    SignUpInfo signup_info = 4;
+
+    // The current chain head’s block num. This is used as a form of time
+    // to help with the enforcement of the C and K tests
+    int32 block_num = 5:
```

Comment:
Same comment as above with regard to K.

Also, the transaction payload contains the block number?  Isn't this the block number when the transaction was created, not the block number that contains this transaction when the block was committed?  Technically the C policy uses the block number for when the validator information was added/updated via the transaction being included in a committed block.